optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/InnerConstraintFactory.java

```diff
@@ -82,22 +81,11 @@
 
     public <A> void assertValidFromType(Class<A> fromType) {
         SolutionDescriptor<Solution_> solutionDescriptor = getSolutionDescriptor();
-        Stream<Class> entityClassStream = solutionDescriptor.getEntityDescriptors().stream()
+        Stream<Class<?>> entityClassStream = solutionDescriptor.getEntityDescriptors().stream()
                 .map(EntityDescriptor::getEntityClass);
```

Comment:
Maybe we don't need a field `problemFactClassSet` on SolutionDescriptor, but a `problemFactOrEntityClassSet` (== allAcceptedClassSet) field instead? That would avoid this mapping too and the problemFactOrEntityClassSet is only used here.